### PR TITLE
cli Test trashbin

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
Add cli tests for trashbin
Related issue: #33536

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by paurakhsharma** [comment]
@individual-it, I have addressed all the comments. Please review

**Comment by paurakhsharma** [comment]
@phil-davis, As you can see in the test when we run occ command `trashbin:cleanup` all users trashbin are cleaned. Isn't this risky? because the administrator can type this by accident. There should be some verification or `--all` should be added. What do you think? Or do we think administrator are not supposed to make mistakes?

**Review by individual-it — APPROVED** [review]

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 1]
```diff
@@ -819,6 +819,28 @@ public function theAdministratorDeletesSystemConfigKeyUsingTheOccCommand($key) {
 		);
 	}
 
+	/**
+	 * @When the administrator empties the trasbin of user :user using the occ command
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: s/trasbin/trashbin

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 2]
```diff
@@ -819,6 +819,28 @@ public function theAdministratorDeletesSystemConfigKeyUsingTheOccCommand($key) {
...
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function theAdministratorEmptiesTheTrasbinOfUserUsingTheOccCommand($user) {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: s/trasbin/trashbin

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 3]
```diff
@@ -819,6 +819,28 @@ public function theAdministratorDeletesSystemConfigKeyUsingTheOccCommand($key) {
...
+		$this->featureContext->invokingTheCommand(
+			"trashbin:cleanup $user"
+		);
+	}
+
+	/**
+	 * @When the administrator empties the trasbin of all users using the occ command
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: s/trasbin/trashbin

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 4]
```diff
@@ -819,6 +819,28 @@ public function theAdministratorDeletesSystemConfigKeyUsingTheOccCommand($key) {
...
+	 *
+	 * @return void
+	 */
+	public function theAdministratorEmptiesTheTrasbinOfAllUsersUsingTheOccCommand() {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: s/trasbin/trashbin

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 5]
```diff
@@ -819,6 +819,28 @@ public function theAdministratorDeletesSystemConfigKeyUsingTheOccCommand($key) {
...
+		$this->theAdministratorEmptiesTheTrasbinOfUserUsingTheOccCommand('');
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: s/trasbin/trashbin

File: tests/acceptance/features/cliTrashbin/trashbin.feature [hunk 6]
```diff
@@ -0,0 +1,30 @@
+@cli @TestAlsoOnExternalUserBackend @files_trashbin-app-required
+Feature: files and folders can be deleted from the trashbin
+  As an admin
+  I want to delete files and folders from the trashbin
+  So that I can control user trashbin space and which files are kept in that space
+
+  Scenario: delete files and folder of a user from a trashbin
```
Comment (phil-davis) [inline tests/acceptance/features/cliTrashbin/trashbin.feature]: s/a trashbin/the trashbin

File: tests/acceptance/features/cliTrashbin/trashbin.feature [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    And user "user0" has deleted file "/textfile1.txt"
+    And user "user0" has deleted folder "/PARENT"
+    When the administrator empties the trasbin of user "user0" using the occ command
+    Then the command should have been successful
+    And the command output should contain the text 'Remove deleted files of   user0'
+    Then as "user0" the file with original path "/textfile0.txt" should not exist in trash
+    And as "user0" the file with original path "/textfile1.txt" should not exist in trash
+    And as "user0" the folder with original path "/PARENT" should not exist in trash
+
+  Scenario: delete files and folder of all user from a trashbin
```
Comment (phil-davis) [inline tests/acceptance/features/cliTrashbin/trashbin.feature]: s/a trashbin/the trashbin

**Review by phil-davis — CHANGES_REQUESTED** [review]

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on: #33548

**Pull request merged**